### PR TITLE
Development

### PR DESCRIPTION
**Pull request by cneker**

### CODE REVIEW

File: InnoGotchi.MVC/InnoGotchi.MVC/Views/Authentication/Registration.cshtml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+﻿@model InnoGotchi.MVC.Models.User.UserForRegistrationDto
+
+@{
+    ViewData["Title"] = "Registration";
+}
+@if (!ViewData.ModelState.IsValid && ViewData.ModelState["Error"].Errors.Count > 0)
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Views/Authentication/Registration.cshtml]: If there are some errors on registration, exception is thrown
image
Model to test: (the password testtest)
image

File: InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        private readonly RequestDelegate _next;
+
+        public AddingAuthenticationHeaderMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            var token = context.Request.Cookies["jwt"];
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs]: Is there a better way? Can you send the token in the header immediately and delete this middleware?

**Review by kseniya-kalita — COMMENTED** [review]
Looks good, just a few things to resolve :)

File: InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        private (string, string) GetRequiredParameters()
+        {
+            var jwt = Request.Cookies["jwt"];
+            var userId = User.Claims.First(c => c.Type == "Id").Value;
+            return (jwt, userId);
+        }
+
+        public async Task<IActionResult> FarmOverview()
```
Comment (kseniya-kalita) [inline InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs:32]: It's better to add HttpPut/HttpPost, etc. attributes to the endpoints

File: InnoGotchi.MVC/InnoGotchi.MVC/Views/Authentication/Registration.cshtml (hunk 1 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Views/Authentication/Registration.cshtml]: fixed

File: InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs (hunk 3 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Controllers/FarmController.cs:32]: added

File: InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs (hunk 2 above)
Comment (cneker) [reply InnoGotchi.MVC/InnoGotchi.MVC/Middlewares/AddingAuthenticationHeaderMiddleware.cs]: I didn't find a better way, but I can call MVC project by jquery and then pass token in header. I don't know another way how to add header to request. Should I use jquery to all my calls to mvc?

**Review by kseniya-kalita — COMMENTED** [review]
Looks good, MVC is accepted!

**Pull request merged**